File: cpp/168_Excel Sheet Column Title.cpp
class Solution {
public:
	string convertToTitle(int n) {
		if (n == 0) {
			return "";
		}
		else {
			return convertToTitle((n - 1) / 26) + (char)('A' + (n - 1) % 26);
		}
	}
};

// iterative:
class Solution {
public:
	string convertToTitle(int n) {
		string res;
		while (n) {
			n--;
			res += 'A' + n % 26;
			n /= 26;
		}
		return res;
	}
};

File: java/168.excel-sheet-column-title.java
/*
 * @lc app=leetcode id=168 lang=java
 *
 * [168] Excel Sheet Column Title
 */
class Solution {
    public String convertToTitle(int n) {
        if (n == 0)
            return "";
        return convertToTitle((n - 1) / 26) + (char) ('A' + (n - 1) % 26);
    }
}
